src/copy_from_template.py

```diff
@@ -117,6 +128,66 @@ def get_folder(created_entities, folder_name):
     folder = folder_finder.pop()
     return folder["entity"]
 
+
+def modify_file_view_types(
+        syn, file_view_id, default_str_length=128,
+        xl_str_fields=["clientInfo"], xl_str_length=512):
+    '''
+    Some string values in the file view are bound to become longer
+    than the default set upon creation of the view. We expand their length
+    to a conservative value so that we can be reasonably certain the file
+    view won't break at some point in the future. We also correct the types
+    of other columns.
+    '''
+    ignore_cols = ["name", "etag", "type"]
+    date_cols = ["exportedOn", "eventTimestamp", "uploadedOn", "scheduleModifiedOn"]
+    boolean_cols = ["timeWindowPersistent"]
+    int_cols = ["sessionInstanceStartDay", "sessionInstanceEndDay",
+            "assessmentRevision", "participantVersion"]
+    file_view = syn.get(file_view_id)
+    cols = list(syn.getColumns(file_view["columnIds"]))
+    col_changes = []
+    for c in cols:
+        if c["columnType"] == "STRING" and c["name"] not in ignore_cols:
+            new_col = deepcopy(c)
+            new_col.pop("id")
+            if new_col["name"] in date_cols:
+                new_col["columnType"] = "DATE"
+                new_col["maximumSize"] = None
+            elif new_col["name"] in boolean_cols:
+                new_col["columnType"] = "BOOLEAN"
+                new_col["maximumSize"] = None
+            elif new_col["name"] in int_cols:
+                new_col["columnType"] = "INTEGER"
+                new_col["maximumSize"] = None
+            elif new_col["name"] in xl_str_fields:
+                new_col["maximumSize"] = xl_str_length
+            else:
+                new_col["maximumSize"] = default_str_length
```

Comment:
Did we want to eventually support the automatic configuration of max str length from the parsing the data itself for the max length?

Reply:
We specify `addAnnotationColumsn` when we create the file view: https://github.com/Sage-Bionetworks/bridge-analytics-template/blob/main/synapseformation_templates/default.yaml#L10

This automatically chooses max length based on the annotations values length. But this is a script to templatize a project, it's pretty early on in data collection, and another value is bound to come along later that is longer than these automatically configured values. So the best we can do is guess what the max length of these values will be, while balancing the table width limitiations.